tests/unit/test_mongodb_backups.py

```diff
@@ -0,0 +1,532 @@
+    @patch("charm.snap.SnapCache")
+    @patch("charm.MongoDBBackups._set_config_options")
+    def test_s3_credentials_set_pbm_failure(self, _set_config_options, snap):
+        """Test charm goes into blocked state when setting pbm configs fail."""
+        mock_pbm_snap = mock.Mock()
+        mock_pbm_snap.present = True
+        mock_pbm_snap.set = mock.Mock()
+        snap.return_value = {"percona-backup-mongodb": mock_pbm_snap}
+        _set_config_options.side_effect = SetPBMConfigError
+        self.harness.charm.app_peer_data["db_initialised"] = "True"
+
+        # triggering s3 event with correct fields
+        mock_s3_info = mock.Mock()
+        mock_s3_info.return_value = {"access-key": "noneya", "secret-key": "business"}
+        self.harness.charm.backups.s3_client.get_s3_connection_info = mock_s3_info
+        relation_id = self.harness.add_relation(RELATION_NAME, "s3-integrator")
+        self.harness.add_relation_unit(relation_id, "s3-integrator/0")
+        self.harness.update_relation_data(
+            relation_id,
+            "s3-integrator/0",
+            {"bucket": "hat"},
+        )
+
+        self.assertTrue(isinstance(self.harness.charm.unit.status, BlockedStatus))
+
+    @patch_network_get(private_address="1.1.1.1")
+    @patch("charm.snap.SnapCache")
+    @patch("charm.MongoDBBackups._set_config_options")
+    @patch("charm.MongoDBBackups._resync_config_options")
+    @patch("ops.framework.EventBase.defer")
+    def test_s3_credentials_config_error(self, defer, resync, _set_config_options, snap):
+        """Test charm defers when more time is needed to sync pbm."""
+        mock_pbm_snap = mock.Mock()
+        mock_pbm_snap.present = True
+        mock_pbm_snap.set = mock.Mock()
+        snap.return_value = {"percona-backup-mongodb": mock_pbm_snap}
+        self.harness.charm.app_peer_data["db_initialised"] = "True"
+        resync.side_effect = SetPBMConfigError
+
+        # triggering s3 event with correct fields
+        mock_s3_info = mock.Mock()
+        mock_s3_info.return_value = {"access-key": "noneya", "secret-key": "business"}
+        self.harness.charm.backups.s3_client.get_s3_connection_info = mock_s3_info
+        relation_id = self.harness.add_relation(RELATION_NAME, "s3-integrator")
+        self.harness.add_relation_unit(relation_id, "s3-integrator/0")
+        self.harness.update_relation_data(
+            relation_id,
+            "s3-integrator/0",
+            {"bucket": "hat"},
+        )
+
+        self.assertTrue(isinstance(self.harness.charm.unit.status, BlockedStatus))
+
+    @patch_network_get(private_address="1.1.1.1")
+    @patch("charm.snap.SnapCache")
+    @patch("charm.MongoDBBackups._set_config_options")
+    @patch("charm.MongoDBBackups._resync_config_options")
+    @patch("ops.framework.EventBase.defer")
+    def test_s3_credentials_syncing(self, defer, resync, _set_config_options, snap):
+        """Test charm defers when more time is needed to sync pbm."""
```

Comment:
Is the description of this function  and the following description correct ? I believe a more specific description will help to quickly understand the purpose of the test.

Reply:
good catch! Must have been a copy paste error I will update